base/floatfuncs.jl

```diff
@@ -105,24 +105,30 @@ end
 
 function _signif_og(x, digits, base)
     if base == 10
-        oftype(x, 10. ^ floor(log10(abs(x)) - digits + 1.))
+        e = floor(log10(abs(x)) - digits + 1.)
+        og = oftype(x, exp10(abs(e)))
     elseif base == 2
-        oftype(x, 2. ^ floor(log2(abs(x)) - digits + 1.))
+        e = exponent(abs(x)) - digits + 1.
+        og = oftype(x, exp2(abs(e)))
```

Comment:
Should there be any difference in performance in generic exponentation whether the exponent is an integer or a float? I thought having an integer exponent might be faster, but my small test shows no significant difference.
If it makes no difference, we could have `e` always be an Int and use `ldexp`. So the only change would be the deletion of the `.` and then we'd have a type stable `e`?